pandas/core/reshape/pivot.py

```diff
@@ -159,15 +159,15 @@ def pivot_table(data, values=None, index=None, columns=None, aggfunc='mean',
     if isinstance(table, DataFrame):
         table = table.sort_index(axis=1)
 
-    if fill_value is not None:
-        table = table.fillna(value=fill_value, downcast='infer')
-
     if margins:
         if dropna:
```

Comment:
You're definitely right. I think it should be (if `dropna=False`):


| b       | 3           | 4  | All|
| ------------- |:-------------:| -----:|-----:|
|a|||
|1.0|1|0|1|
|2.0|1|3|4|
|np.nan|0|1|1|
|All|2|4|6|

- I need to fix the np.nan as it is still being ignored even with the current fix when `dropna=False` (i.e. the fix only works for `None`)
- Actually I didn't get why removing `dropna` here would help yet. I'll check closer.